docs/projects/cn0577: Migrate documentation (#1363)
analogdevicesinc/hdl: docs/projects/cn0577/index.rst

==================== ===============

I2C connections
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. list-table::
   :widths: 20 20 20 20 20
   :header-rows: 1

   * - I2C type
     - I2C manager instance
     - Alias
     - Address
     - I2C subordinate
   * - PL
     - iic_fmc
     - axi_iic_fmc
     - 0x4162_0000
     - ---
   * - PL
     - iic_main
     - axi_iic_main
     - 0x4160_0000
     - ---

GPIOs
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. list-table::
   :widths: 25 20 20 20 15
   :header-rows: 2

   * - GPIO signal
     - Direction
     - HDL GPIO EMIO
     - Software GPIO
     - Software GPIO
   * -
     - (from FPGA view)
     -
     - Zynq-7000
     - Zynq MP
   * - testpat_cntrl
     - IN
     - 32
     - 86
     - 110
   * - pd_cntrl
     - IN
     - 33
     - 87
     - 111

Interrupts
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Below are the Programmable Logic interrupts used in this project.

================ === ========== ===========
Instance name    HDL Linux Zynq Actual Zynq
================ === ========== ===========

> ad_cpu_interrupt 13  57         89

Reviewer: The name of the instance is with this interrupt is `axi_ltc2387_dma`

================ === ========== ===========

Building the HDL project
-------------------------------------------------------------------------------

The design is built upon ADI's generic HDL reference design framework.

> ADI does not distribute the bit/elf files of these projects so they
> must be built from the sources available :git-hdl:`here <>`. To get
> the source you must

Reviewer: ```suggestion
ADI distributes the bit/elf files of these projects as part of the
:dokuwiki:`ADI Kuiper Linux <resources/tools-software/linux-software/kuiper-linux>`.
If you want to build the sources, ADI makes them available on the
:git-hdl:`HDL repository </>`. To get the source you must
```

`clone <https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository>`__
the HDL repository.

> Default:

Reviewer: ```suggestion
Default (two-lane mode):
```

.. code-block::
   :linenos:

   user@analog:~$ cd hdl/projects/cn0577/zed
   user@analog:~/hdl/projects/cn0577/zed$ make

> If one-lane output is desired:

Reviewer: ```suggestion
If one-lane mode is desired:
```

.. _cn0577:

CN0577 HDL project
================================================================================

Overview
-------------------------------------------------------------------------------

> The :adi:`EVALCN0577-FMCZ <CN0577>` provides an analog front-end and an FMC

Reviewer: ```suggestion
The :adi:`CN0577` provides an analog front-end and an FMC
```
It's enough to be referred just like this. Should be updated in the whole document

digital interface for :adi:`LTC2387-18`, its core. It is a low noise, high
speed successive approximation register (SAR) ADC with a resolution of 18 bits
and sampling rate up to 15MSPS.

:adi:`EVALCN0577-FMCZ <CN0577>` includes an on-board reference oscillator and a
retiming circuit to minimize signal-to-noise ratio (SNR) degradation due to the
FPGA additive jitter.

In order to support high speed operations while minimizing the number of data
lines, a serial LVDS digital interface is used. It has a one-lane and two-lane
output modes, allowing the user to optimize the interface data rate for each
application, through setting a parameter. Current revision of :adi:`EVALCN0577-FMCZ <CN0577>`
is Rev A. :adi:`EVALCN0577-FMCZ <CN0577>` and :xilinx:`ZedBoard <products/boards-and-kits/1-8dyf-11.html>`
are connected together to build a development system setup.

> Supported boards
> -------------------------------------------------------------------------------
> 
> -  :adi:`CN0577 <CN0577>`
> -  :adi:`LTC2387-18 chip <LTC2387-18>`
> 
> Supported devices
> -------------------------------------------------------------------------------
> 
> -  :adi:`ADAQ23876`
> -  :adi:`ADA4945-1`
> -  :adi:`ADG3241`
> -  :adi:`ADN4661`
> -  :adi:`ADR4520`

Reviewer: "Supported boards" means which eval board works with this HDL project.
"Supported devices" means which is the main chip that is actually used in that eval board.

```suggestion
Supported boards
-------------------------------------------------------------------------------

-  :adi:`CN0577`

Supported devices
-------------------------------------------------------------------------------

-  :adi:`ADAQ23876`
-  :adi:`LTC2387-18`
```

Supported carriers
-------------------------------------------------------------------------------

.. list-table::
   :widths: 35 35 30
   :header-rows: 1

   * - Evaluation board
     - Carrier
     - FMC slot
   * - :adi:`CN0577 <CN0577>`
     - :xilinx:`ZedBoard <products/boards-and-kits/1-8dyf-11.html>`

>      - FMC-LPC

Reviewer: No need for a table, since it's just a carrier, you can make it as a list. Check other projects as an example

Block design
-------------------------------------------------------------------------------

.. warning::
    The VADJ for the Zedboard must be set to 2.5V.

Configuration modes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 - TWOLANES: specifies the number of lanes used

    - 1 - two-lane output mode (default)
    - 0 - one-lane output mode  

> 

Reviewer: Should be moved at line 78

The data path and clock domains are depicted in the below diagram:

.. image:: ../cn0577/cn0577_zed_block_diagram.svg
   :width: 800
   :align: center
   :alt: CN0577/ZedBoard block diagram

Configuration modes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

tbc

Jumper setup
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Depending on what configuration of pins is chosen on the jumpers P1, P2 and P3,
the device can act in different modes, as described below. Of course, the PD
jumper overrides the PD signal from the FPGA. It is controlled by a
one-bit-adc-dac, in software.

  - P1 - configures PD_N

    - Shorting pins 1 and 2 → PD_N = 1, device is not powered down
    - Shorting pins 2 and 3 → PD_N = 0, device is powered down
  
  - P2 - configures TESTPAT

    - Shorting pins 1 and 2 → TESTPAT = 1, pattern testing is active
    - Shorting pins 2 and 3 → TESTPAT = 0, pattern testing is inactive

  - P3 - configures TWOLANES parameter

    - Shorting pins 1 and 2 → TWOLANES = 1 (TWO LANES mode)
    - Shorting pins 2 and 3 → TWOLANES = 0 (ONE LANE mode)

> 
> The FMC connector connects to the LPC connector of the carrier board. For more
> detalis, check :adi:`EVALCN0577-FMCZ <CN0577>` datasheet.
> 

Reviewer: ```suggestion
```

Clock scheme
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-  The clock architecture of the :adi:`EVALCN0577-FMCZ <CN0577>` is designed
   with careful consideration to ensure low jitter and low phase noise

> 

Reviewer: Remove the empty lines from between the list's items

-  An on-board 120 MHz voltage controlled crystal oscillator (VCXO) is used to
   provide the clock for the :adi:`EVALCN0577-FMCZ <CN0577>` board and the FPGA.
   It is further named as reference clock. This clock is gated and fed back to
   the device as the sampling clock, on which the data was sampled

-  The DMA runs on the ZynqPS clock FCLK_CLK0 which has a frequency of 100MHz

> For more detalis, check :adi:`EVALCN0577-FMCZ <CN0577>` datasheet.

Reviewer: ```suggestion
```

CPU/Memory interconnects addresses
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The addresses are dependent on the architecture of the FPGA, having an offset
added to the base address from HDL (see more at :ref:`architecture`).

==================== ===============
Instance             Zynq/Microblaze
==================== ===============
axi_ltc2387          0x44A0_0000
axi_ltc2387_dma      0x44A3_0000
axi_pwm_gen          0x44A6_0000
==================== ===============

> I2C connections

Reviewer: To be removed

Author: Again, followed CN0585 doc model I2C section. Should I remove from there too?

Reviewer: This section should be removed because there are no project-specific I2C connections. These connections are specific to the carrier

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. list-table::
   :widths: 20 20 20 20 20
   :header-rows: 1

   * - I2C type
     - I2C manager instance
     - Alias
     - Address
     - I2C subordinate
   * - PL
     - iic_fmc
     - axi_iic_fmc
     - 0x4162_0000
     - ---
   * - PL
     - iic_main
     - axi_iic_main
     - 0x4160_0000
     - ---

GPIOs
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. list-table::
   :widths: 25 20 20 20 15
   :header-rows: 2

   * - GPIO signal
     - Direction
     - HDL GPIO EMIO
     - Software GPIO
     - Software GPIO
   * -
     - (from FPGA view)
     -
     - Zynq-7000
     - Zynq MP
   * - pd_cntrl
     - IN
     - 32
     - 86
     - 110
   * - testpat_cntrl
     - IN
     - 33
     - 87

>      - 111

Reviewer: PD is 33 in HDL and TESTPAT is 32. The other columns should be updated as well.

Interrupts
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Below are the Programmable Logic interrupts used in this project.

================ === ========== ===========
Instance name    HDL Linux Zynq Actual Zynq
================ === ========== ===========
ad_cpu_interrupt 13  57         89
================ === ========== ===========

Building the HDL project
-------------------------------------------------------------------------------

The design is built upon ADI's generic HDL reference design framework.
ADI does not distribute the bit/elf files of these projects so they
must be built from the sources available :git-hdl:`here <>`. To get
the source you must
`clone <https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository>`__
the HDL repository.

.. code-block::
   :linenos:

   user@analog:~$ cd hdl/projects/cn0577/zed

>    user@analog:~/hdl/projects/cn0577/zed$ make

Reviewer: Give an example on how to use the parameter:
```suggestion
   user@analog:~/hdl/projects/cn0577/zed$ make TWOLANES=0
```

Author: Should I add this as a second example? Or follow the example as it is in AD4630-FMC documentation?

Reviewer: Ok, add it as a second example, and specify like a comment, that when running "make", the default configuration is ...

A more comprehensive build guide can be found in the :ref:`build_hdl` user
guide.

Resources
-------------------------------------------------------------------------------

Systems related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-  :dokuwiki:`[Wiki] EVAL-CN0577-FMCZ User Guide <resources/eval/user-guides/circuits-from-the-lab/cn0577>`
-  :dokuwiki:`[Wiki] CN0577 HDL Reference Design <resources/eval/user-guides/circuits-from-the-lab/cn0577/hdl>`

Hardware related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-  Product datasheets:

   -  :adi:`ADAQ23876`
   -  :adi:`ADR4520`
   -  :adi:`ADA4945-1`
   -  :adi:`ADN4661`

> -  `Circuit Note CN-0577 <https://www.analog.com/media/en/reference-design-documentation/reference-designs/cn0577.pdf>`__

Reviewer: Here should be only the LTC2387 mentioned

HDL related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-  :git-hdl:`CN0577 HDL project source code <projects/cn0577>`

.. list-table::
   :widths: 30 35 35
   :header-rows: 1

   * - IP name
     - Source code link
     - Documentation link
   * - AXI_LTC2387
     - :git-hdl:`library/axi_ltc2387`

>      - :dokuwiki:`[Wiki] <resources/fpga/docs/axi_ltc2387>`

Reviewer: All the links from dokuwiki should be changed to :ref: because we already have them ported

Author: Doing this the links won't work anymore. I did change only for AXI_DMAC and AXI_PWM_GEN. If I change the other links, they won't work. Is there a new update to the tags?

Author: Changed the labels to :ref: of the items that are already ported.

   * - AXI_DMAC
     - :git-hdl:`library/axi_dmac`
     - :ref:`here <axi_dmac>`
   * - AXI_SYSID
     - :git-hdl:`library/axi_sysid`
     - :dokuwiki:`[Wiki] <resources/fpga/docs/axi_sysid>`
   * - SYSID_ROM
     - :git-hdl:`library/sysid_rom`
     - :dokuwiki:`[Wiki] <resources/fpga/docs/axi_sysid>`
   * - AXI_CLKGEN
     - :git-hdl:`library/axi_clkgen`
     - :dokuwiki:`[Wiki] <resources/fpga/docs/axi_clkgen>`
   * - AXI_PWM_GEN
     - :git-hdl:`library/axi_pwm_gen`
     - :ref:`here <axi_pwm_gen>`
   * - AXI_HDMI_TX
     - :git-hdl:`library/axi_hdmi_tx`
     - :dokuwiki:`[Wiki] <resources/fpga/docs/axi_hdmi_tx>`
   * - AXI_I2S_ADI
     - :git-hdl:`library/axi_i2s_adi`
     - —
   * - AXI_SPDIF_TX
     - :git-hdl:`library/axi_spdif_tx`
     - 	—
   * - UTIL_I2C_MIXER
     - :git-hdl:`library/util_i2c_mixer`
     - 	—

Software related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> -  :dokuwiki:`[Wiki] LTC2387 SAR ADC IIO Linux driver page <resources/tools-software/linux-drivers/iio-adc/ltc2387>`

Reviewer: Link the driver and the dts as well

Author: Followed the model present in CN0585 documentation. Should I change there too? If so, should I make a new PR?

> -  Python support (THIS IS JUST AN EXAMPLE):
> 
>    -  `PyADI-IIO documentation <https://analogdevicesinc.github.io/pyadi-iio/>`__

Reviewer: ```suggestion
```

.. _cn0577:

CN0577 HDL project
================================================================================

Overview
-------------------------------------------------------------------------------

The :adi:`EVALCN0577-FMCZ <CN0577>` provides an analog front-end and an FMC 
digital interface for :adi:`LTC2387-18`, its core. It is a low noise, high 
speed successive approximation register (SAR) ADC with a resolution of 18 bits 

> and sampling rate up to 15MSPS. :adi:`EVALCN0577-FMCZ <CN0577>` includes an 

Reviewer 2: Add two line break after `15MBPS.` to generate a new paragraph.

on-board reference oscillator and a retiming circuit to minimize signal-to-noise 

> ratio (SNR) degradation due to the FPGA additive jitter. In order to support 

Reviewer 2: Add two line break after `jitter.` to generate a new paragraph.

digital interface is used. It has a one-lane and two-lane output modes, 
allowing the user to optimize the interface data rate for each application, 
through setting a parameter.
Current revision of :adi:`EVALCN0577-FMCZ <CN0577>` is Rev A.
:adi:`EVALCN0577-FMCZ <CN0577>` and :xilinx:`ZedBoard <products/boards-and-kits/1-8dyf-11.html>` 
are connected together to build a development system setup.

Supported boards
-------------------------------------------------------------------------------

-  :adi:`CN0577 <CN0577>`
-  :adi:`LTC2387-18 chip <LTC2387-18>`

Supported devices
-------------------------------------------------------------------------------

-  :adi:`ADAQ23876`
-  :adi:`ADA4945-1`
-  :adi:`ADG3241`
-  :adi:`ADN4661`
-  :adi:`ADR4520`

Supported carriers
-------------------------------------------------------------------------------

.. list-table::
   :widths: 35 35 30
   :header-rows: 1

   * - Evaluation board
     - Carrier
     - FMC slot
   * - :adi:`CN0577 <CN0577>`
     - :xilinx:`ZedBoard <products/boards-and-kits/1-8dyf-11.html>`
     - FMC-LPC

Block design
-------------------------------------------------------------------------------

> The architecture is composed from one :adi:`ADA4945-1` fully differential 

Reviewer 2: Not only the ADA4945-1 but also the other supported devices, right?

Author: I realized this info is not relevant to the HDL project. I followed the example given in the same section of CN0855's documentation before. This section will be removed.

analog-to-digital driver interface IP that is driving the :adi:`LTC2387-18` 
which is an 18 bits analog-to-digital converter interface IP. All these IPs 
utilize an 120MHz reference clock, which is produced by an axi_clkgen IP.

Before proceeding with the project, please pay attettion to this pieces of
infomation related to the HDL project.

.. note::

>     In the main version, the TWOLANES parameter is set to 1, so it works 

Reviewer 2: Remove trailing whitespace

>     In the main version, the TWOLANES parameter is set to 1, so it works 

Reviewer 2: the axi_ltc2387 parameter is called TWOLANES, but the config var in zed/system_bd.tcl is named `two_lanes`, which might be confusing.

Author: Should this change be made in another PR?

Reviewer 2: Yes, open other PR for this change

    only in two-lane output mode.

.. warning::
    The VADJ for the Zedboard must be set to 2.5V.

> 

Reviewer 2: Missing Parameters section, explaining TWOLANES

Block diagram
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The data path and clock domains are depicted in the below diagram:

.. image:: ../cn0577/cn0577_zed_block_diagram.svg
   :width: 800
   :align: center
   :alt: CN0577/ZedBoard block diagram

Jumper setup
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Depending on what configuration of pins is chosen on the jumpers P1, P2 and P3, 
the device can act in different modes, as described below. Of course, the PD 
jumper overrides the PD signal from the FPGA. It is controlled by a 
one-bit-adc-dac, in software.

> .. list-table::

Reviewer 2: Better to have this as a list, not a list table
